### PR TITLE
Complete Project

### PR DESCRIPTION
**Pull request by Kwaku28**
### Features of the app include:
1. Create a new account
2. Log in to an existing account
3. Sign out from a session
4. Create a Budget(s)
5. Add expenses to a budget
6. Check the list of expenses for a particular budget
7. Display the total amount of expenses for a particular budget
8. Dynamic navigation bar

#### Tasks completed include the following:
- [x] Interactions
    - Splash screen
       - A simple page with the name of the app and links to the sign-up and log-in pages.
   
    - Sign up and login pages
       - The user is able to register in the app with full name, email, and password (all mandatory).
       - The user can log into the app using email and password.
    
    - Home page (budget page)
       - The user is presented with the budget page after login in.
       - For each budget, the user can see their name, icon, and the total amount of all the expenses that belong to that budget.
       - When the user clicks (or taps) on a budget item, the application navigates to the transactions page for that budget.
       - Added a button to add a new budget.
      
    - Expenses page
      - For a given budget, the list of transactions is presented, ordered by the most recent.
      - Added a button to add a new expense.
      - Added a "Back" button (<), that navigates the user to the home page.
   
    - "Add a new category" page
      - Added a form to create a new budget, indicating their name and icon.
      - Added a "Save" button to create the new budget, and the user is taken to the home page on success.
   
    - "Add a new expense" page
      - Added a form to create a new expense with:
        - name (mandatory)
        - amount (mandatory)
        - budget (required at least one)
      - Added a "Save" button to create a new expense.

- [x] Testing
   - Added unit and integration tests.

- [x] Technical
   - Used Postgres as the database.
   - Used devise for authentication.
   - Validates all user input to make sure that anyone with bad intentions cannot compromise the app.
   - Used `.erb` view template engine

Thank you🤝

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,197 @@
...
+- Add expenses to a budget
+- Check the list of expenses for a particular budget
+- Display the total amount of expenses for a particular budget
+- Tests for budgets
+- Tests for expenses
+- Dynamic navbar
+
+# 🚀 Live Demo <a name="live-demo"></a>
+
+👉 [Live Demo](https://grow-rich.onrender.com)<br><br>
```
Comment (DuaneDave) [inline README.md:65]: Good attempt with the deployment, however when I click on the link, it routes to a 502 page. I recommend you look up what is causing the issue so as to make the page accessible.

image

File: spec/features/budgets_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      fill_in 'Password confirmation', with: 'kwaku28'
+      click_button 'Sign up'
+      visit user_session_path
+      fill_in 'Email', with: 'lebron255@gmail'
+      fill_in 'Password', with: 'kwaku28'
+      click_button 'Log in'
+      visit new_budget_path
+      fill_in 'Name', with: 'Food'
+      fill_in 'Icon URL', with: 'https://99designs-blog.imgix.net/blog/wp-content/uploads/2017/08/BRAND-YOURSELF-700x420.jpg?auto=format&q=60&fit=max&w=930'
+      click_button 'Save'
```
Comment (DuaneDave) [inline spec/features/budgets_spec.rb:19]: When I run the tests, I get a whole bunch of errors on the console. I suggest resolving any issues that might be going on with the test in order to prepare your work for approval.

image

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @Kwaku28 👋, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (Kwaku28) [reply README.md:65]: Link fixed

File: app/views/budgets/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+          <p><%= "$#{budget.expenses.sum(:amount)}" %></p>
+        </div>
+      </div>
+    <% end %>
+  </div>
+  <div class="plus d-flex justify-content-end p-5">
+    <%= link_to new_budget_path do %>
+      <i class="bi bi-plus-circle-fill"></i>
+    <% end %>
+  </div>
```
Comment (MahmoudBakr23) [inline app/views/budgets/index.html.erb:24]: - [ ] When there are no budgets created yet, the home page is just empty and there are no navigation options unless I manually update the URL. Please fix this issue and update it on the Live Demo Link of the app. 🙏 
GrowRich

- [ ] Also, the design is way too different from the one you presented in the video. Please inspect the issue on the deployment and fix it. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hi @Kwaku28 😃, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but some are still not been addressed yet.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/budgets/index.html.erb (hunk 3 above)
Comment (Kwaku28) [reply app/views/budgets/index.html.erb:24]: - [ ] Splash screen
Splash

- [ ] Home screen
home

File: spec/features/budgets_spec.rb (hunk 2 above)
Comment (AmaduKamara) [inline spec/features/budgets_spec.rb:19]: - [ ]Good job working on some of the test files, but there are a few tests failing and crashing the simulation test when running. Kindly work on the remaining failed test and your app should be on its way to approval 👍🏽

Screenshot 2023-08-11 at 3 55 13 PM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️ 

Hi @Kwaku28, 

Good job so far on working on the changes highlighted by the previous reviewers 👍🏽
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks pass successfully 👍🏽
✅ The app runs fine on the local and online servers as expected 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Good job so far on your project. There are a few changes to be made to make your app looks nice as required. From the design point of view, you only have to reduce the font sizes of texts, adjust the navbar to position at the top of the screen instead of hanging, and reduce the plus(+) icon so the interface looks nice on a mobile screen just as is in the design used for inspiration to build this app. 

Screenshot 2023-08-11 at 3 53 39 PM
Screenshot 2023-08-11 at 3 51 53 PM
Screenshot 2023-08-11 at 3 52 48 PM


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/features/budgets_spec.rb (hunk 2 above)
Comment (Kwaku28) [reply spec/features/budgets_spec.rb:19]: test cases

**Review by cynthiainga — APPROVED** [review]
## STATUS APPROVED ✔️ 🎉 ✔️ 

Hi @Kwaku28, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] The app works as expected ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**